### DISCUSSION TITLE
BootstrapVue template doesn't work with screen's width <576px

### DISCUSSION THREAD

**Discussion by tesicg** [bootstrap-vue/bootstrap-vue#6391] [Q&A] [answered]
The code looks as following:

    <template lang="pug">
    b-container
      b-row
        b-col
          h1
            strong LICOTA® — ПРОФЕССИОНАЛЬНЫЙ ИНСТРУМЕНТ ВЫСШЕГО КЛАССА ДЛЯ АВТОСЕРВИСОВ И ПРОМЫШЛЕННЫХ ПРЕДПРИЯТИЙ
      b-row.mt-1
        b-col(class='col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column')
          b-card(img-src='https://licota.ru/home_banners/info-item-1.png' img-alt='Image' img-top tag='article' class='info-block__card--no-border h-100')
            b-card-text Имеет TUV сертификат ISO 9001:2008. Соответствует требованиям GS, CE, EAC и превышает нормы DIN и ГОСТ
        b-col(class='col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column')
          b-card(img-src='https://licota.ru/home_banners/info-item-2.png' img-alt='Image' img-top tag='article' class='info-block__card--no-border h-100')
            b-card-text Один из ведущих производителей инструмента в мире с трехуровневым контролем качества
        b-col(class='col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column')
          b-card(img-src='https://licota.ru/home_banners/info-item-3.png' img-alt='Image' img-top tag='article' class='info-block__card--no-border h-100')
            b-card-text Инструмент адаптирован к российским условиям и имеет пожизненную гарантию на брак изготовителя
      b-row.mt-1
        b-col(class='col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column')
          b-card(img-src='https://licota.ru/home_banners/info-item-4.png' img-alt='Image' img-top tag='article' class='info-block__card--no-border h-100')
            b-card-text Наши клиенты из автомобильного сегмента, которые выбрали Licota®
        b-col(class='col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column')
          b-card(img-src='https://licota.ru/home_banners/info-item-5.png' img-alt='Image' img-top tag='article' class='info-block__card--no-border h-100')
            b-card-text Эти промышленные предприятия уверенно работают с Licota®
        b-col(class='col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column')
          b-card(img-src='https://licota.ru/home_banners/info-item-6.png' img-alt='Image' img-top tag='article' class='info-block__card--no-border h-100')
            b-card-text Дилеры Licota® по всей России, от Калининграда до Камчатки
      b-row.mt-1
        b-col(class='col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column')
          b-button(class='info-block__button' block size='lg') О ПРОИЗВОДИТЕЛЕ
        b-col(class='col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column')
          b-button(class='info-block__middle-button' block size='lg') КАТАЛОГ ТОВАРОВ
        b-col(class='col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column')
          b-button(class='info-block__button' block size='lg') ГДЕ КУПИТЬ?
    </template>

    <script lang="ts">
    import Vue from 'vue'
    import { Data } from 'types/header/nav/catalog'
    import testData from '../data/testData.json'

    export default Vue.extend({
      data: (): Data => ({
        catalog: [],
      }),
      mounted(): void {
        this.catalog = testData.catalog
      },
    })
    </script>

    <style lang="sass">
    .info-block
      font-size: 16px
      text-align: center
    .info-block__column
      padding: 10px
    .info-block__card--no-border
      border: none
    .info-block__button
      background-color: #eee
      background-image: -webkit-linear-gradient(bottom, #ccc 0%, #eee 100%)
      border: 1px solid #cccccc
      color: black
      font-size: 18px
      font-weight: bold
      line-height: 2
      text-shadow: 0.5px 0.866px 0 white
    .info-block__middle-button
      background-color: #007dfa
      background-image: -webkit-linear-gradient(bottom, #00468c 0%, #007dfa 100%)
      border: 1px solid rgb(0, 70, 140)
      color: white
      font-size: 18px
      font-weight: bold
      line-height: 2
      text-shadow: 0.5px 0.866px 0 rgb(0, 30, 60)
    </style>

As you can see, it's Vue.js application where we use "pug" template engine and BootstrapVue component library. I'm trying to make the app responsive using b-col with Bootstrap's col-xs-*, col-sm-*, and so on.

The main page with screen's width <576px looks as following:

[![enter image description here][7]][7]

  [7]: https://i.stack.imgur.com/hVc7F.jpg

UPDATE:

This is code that works fine except for screen's width <576px.

    <template lang="pug">
      b-container
        b-row
          b-col
            h1
              strong LICOTA® — ПРОФЕССИОНАЛЬНЫЙ ИНСТРУМЕНТ ВЫСШЕГО КЛАССА ДЛЯ АВТОСЕРВИСОВ И ПРОМЫШЛЕННЫХ ПРЕДПРИЯТИЙ
        b-row.mt-1
          b-col.col-xs-12.col-sm-12.col-md-6.col-lg-4.col-xl-4.info-block__column
            b-card(img-src='https://licota.ru/home_banners/info-item-1.png' img-alt='Image' img-top tag='article').info-block__card--no-border.h-100
              b-card-text Имеет TUV сертификат ISO 9001:2008. Соответствует требованиям GS, CE, EAC и превышает нормы DIN и ГОСТ
          b-col.col-xs-12.col-sm-12.col-md-6.col-lg-4.col-xl-4.info-block__column
            b-card(img-src='https://licota.ru/home_banners/info-item-2.png' img-alt='Image' img-top tag='article').info-block__card--no-border.h-100
              b-card-text Один из ведущих производителей инструмента в мире с трехуровневым контролем качества
          b-col.col-xs-12.col-sm-12.col-md-6.col-lg-4.col-xl-4.info-block__column
            b-card(img-src='https://licota.ru/home_banners/info-item-3.png' img-alt='Image' img-top tag='article').info-block__card--no-border.h-100
              b-card-text Инструмент адаптирован к российским условиям и имеет пожизненную гарантию на брак изготовителя
          b-col.col-xs-12.col-sm-12.col-md-6.col-lg-4.col-xl-4.info-block__column
            b-card(img-src='https://licota.ru/home_banners/info-item-4.png' img-alt='Image' img-top tag='article').info-block__card--no-border.h-100
              b-card-text Наши клиенты из автомобильного сегмента, которые выбрали Licota®
          b-col.col-xs-12.col-sm-12.col-md-6.col-lg-4.col-xl-4.info-block__column
            b-card(img-src='https://licota.ru/home_banners/info-item-5.png' img-alt='Image' img-top tag='article').info-block__card--no-border.h-100
              b-card-text Эти промышленные предприятия уверенно работают с Licota®
          b-col.col-xs-12.col-sm-12.col-md-6.col-lg-4.col-xl-4.info-block__column
            b-card(img-src='https://licota.ru/home_banners/info-item-6.png' img-alt='Image' img-top tag='article').info-block__card--no-border.h-100
              b-card-text Дилеры Licota® по всей России, от Калининграда до Камчатки
        b-row.mt-1
          b-col.col-xs-12.col-sm-12.col-md-6.col-lg-4.col-xl-4.info-block__column
            b-button.btn.btn-lg.btn-block.info-block__button О ПРОИЗВОДИТЕЛЕ
          b-col.col-xs-12.col-sm-12.col-md-6.col-lg-4.col-xl-4.info-block__column
            b-button.btn.btn-lg.btn-block.info-block__middle-button КАТАЛОГ ТОВАРОВ
          b-col.col-xs-12.col-sm-12.col-md-6.col-lg-4.col-xl-4.info-block__column
            b-button.btn.btn-lg.btn-block.info-block__button ГДЕ КУПИТЬ?
    </template>

The compiled HTML code (BootstrapVue) looks as following:

	<b-container>
		<b-row>
			<b-col>
				<h1><strong>LICOTA® — ПРОФЕССИОНАЛЬНЫЙ ИНСТРУМЕНТ ВЫСШЕГО КЛАССА ДЛЯ АВТОСЕРВИСОВ И ПРОМЫШЛЕННЫХ ПРЕДПРИЯТИЙ</strong></h1>
			</b-col>
		</b-row>
		<b-row class="mt-1">
			<b-col class="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column">
				<b-card class="info-block__card--no-border h-100" img-src="https://licota.ru/home_banners/info-item-1.png" img-alt="Image" img-top="img-top" tag="article">
					<b-card-text>Имеет TUV сертификат ISO 9001:2008. Соответствует требованиям GS, CE, EAC и превышает нормы DIN и ГОСТ</b-card-text>
				</b-card>
			</b-col>
			<b-col class="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column">
				<b-card class="info-block__card--no-border h-100" img-src="https://licota.ru/home_banners/info-item-2.png" img-alt="Image" img-top="img-top" tag="article">
					<b-card-text>Один из ведущих производителей инструмента в мире с трехуровневым контролем качества</b-card-text>
				</b-card>
			</b-col>
			<b-col class="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column">
				<b-card class="info-block__card--no-border h-100" img-src="https://licota.ru/home_banners/info-item-3.png" img-alt="Image" img-top="img-top" tag="article">
					<b-card-text>Инструмент адаптирован к российским условиям и имеет пожизненную гарантию на брак изготовителя</b-card-text>
				</b-card>
			</b-col>
			<b-col class="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column">
				<b-card class="info-block__card--no-border h-100" img-src="https://licota.ru/home_banners/info-item-4.png" img-alt="Image" img-top="img-top" tag="article">
					<b-card-text>Наши клиенты из автомобильного сегмента, которые выбрали Licota®</b-card-text>
				</b-card>
			</b-col>
			<b-col class="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column">
				<b-card class="info-block__card--no-border h-100" img-src="https://licota.ru/home_banners/info-item-5.png" img-alt="Image" img-top="img-top" tag="article">
					<b-card-text>Эти промышленные предприятия уверенно работают с Licota®</b-card-text>
				</b-card>
			</b-col>
			<b-col class="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column">
				<b-card class="info-block__card--no-border h-100" img-src="https://licota.ru/home_banners/info-item-6.png" img-alt="Image" img-top="img-top" tag="article">
					<b-card-text>Дилеры Licota® по всей России, от Калининграда до Камчатки</b-card-text>
				</b-card>
			</b-col>
		</b-row>
		<b-row class="mt-1">
			<b-col class="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column">
				<b-button class="btn btn-lg btn-block info-block__button">О ПРОИЗВОДИТЕЛЕ</b-button>
			</b-col>
			<b-col class="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column">
				<b-button class="btn btn-lg btn-block info-block__middle-button">КАТАЛОГ ТОВАРОВ</b-button>
			</b-col>
			<b-col class="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 info-block__column">
				<b-button class="btn btn-lg btn-block info-block__button">ГДЕ КУПИТЬ?</b-button>
			</b-col>
		</b-row>
	</b-container>

How can I fix it?

**Comment by Hiws** [comment, marked as answer]
That's because `col-xs-*` doesn't exist.
It's simply called `col-*`, so if you replace all of those in your code it should work.
<https://getbootstrap.com/docs/4.6/layout/grid/#grid-options>


I'd also recommend you use the props on `<b-col>` instead of manually writing the classes.
`<b-col cols="12" md="6" lg="4" class="info-block__column">` would be all you need.
